### ISSUE TITLE
CSSStyleDeclaration.*** should return only `string` value.

### ISSUE THREAD

**Issue by tetsuharuohzeki** [microsoft/TypeScript-DOM-lib-generator#97]
By the spec, `CSSStyleDeclaration.properttyName` should only return a `string` value, not be `null`.
- https://www.w3.org/TR/cssom-1/#dom-cssstyledeclaration-getpropertyvalue
- https://www.w3.org/TR/cssom-1/#dom-cssstyledeclaration-camel-cased-attribute

But the current code in this repository defines them as `properttyName: string | null`. This is incorrect.

**Comment by zhengbli** [comment]
I don't get your description. `CSSStyleDeclaration` doesn't have a property called `propertyName`, and your references are hard to follow. Please be clearer.

**Comment by saneyuki** [comment]
@zhengbli

`CSSStyleDeclaration` has many properties which corresponds to CSS's each property (e.g. `CSSStyleDeclaration.width`, `CSSStyleDeclaration.height`, `CSSStyleDeclaration.display`, and etc).

At the current revision, [they are defined as `string | null`](https://github.com/Microsoft/TSJS-lib-generator/blob/42a478a9d8db6ef238578082f28ceaf33d250f2b/baselines/dom.generated.d.ts#L1347).

But by the spec, they don't return `null`, only return `string` value.

- https://www.w3.org/TR/cssom-1/#dom-cssstyledeclaration-getpropertyvalue
- https://www.w3.org/TR/cssom-1/#dom-cssstyledeclaration-camel-cased-attribute

So we need to change type declarations.

**Comment by SaschaNaz** [comment]
I can understand what @saneyuki says, `document.body.style.(someProperty)` always returns string unless `(someProperty)` is unsupported. I can see `string | null` everywhere in lib.d.ts, is it for existence check?

**Comment by SaschaNaz** [comment]
After one year I think think current definition is incorrect. [Per the spec](http://www.w3.org/TR/cssom-1/#dom-cssstyledeclaration-getpropertyvalue):

```webidl
partial interface CSSStyleDeclaration {
  [TreatNullAs=EmptyString] attribute DOMString _camel_cased_attribute;
};
```

It never returns `null`, and if this is really for existence check it should be `undefined`. (I doubt, however. If you do this for the check then why not on others e.g. window.performance?)

**Comment by HolgerJeromin** [comment]
IMO this should be `string` for the common values and `string|undefined` for all vendor prefixed ones (`ms*`, `webkit*`)

**Comment by saschanaz** [comment]
Has been fixed since then.

@github-actions close
